crates/dbsp/src/operator/dynamic/trace.rs

```diff
@@ -569,7 +569,10 @@ where
         );
 
         circuit.cache_insert(TraceId::new(stream.stream_id()), trace.clone());
-        circuit.cache_insert(BoundsId::<T>::new(stream.stream_id()), self.bounds.clone());
+        circuit.cache_insert(
+            BoundsId::<T::Batch>::new(stream.stream_id()),
```

Comment:
very subtle

Reply:
I assume you have checked the other places as well.
why do you need the type in the cache anyway?

Reply:
yes